components/dashboards-web-component/src/designer/components/WidgetConfigurationPanel.jsx

```diff
@@ -106,15 +202,158 @@ class WidgetConfigurationPanel extends React.Component {
         return publishers;
     }
 
+    getPreferences() {
+        let preferences = [];
+        if (!(dashboardLayout.selectedItem && dashboardLayout.selectedItem.config.content[0].props.configs)) {
+            return <div>No Configurable Options</div>;
+        }
+        let widgetId = dashboardLayout.selectedItem.config.content[0].props.id;
+        let dashboard = this.props.getDashboard();
+        let pageId = this.props.getPageId();
+        let page = DashboardUtils.getPage(dashboard, pageId);
+        let widget = DashboardUtils.searchForWidget(page, widgetId);
+        let options = widget.props.configs.options;
+
+        if (this.state.id !== widgetId) {
+            let newState = this.state;
+            newState.options = options;
+            newState.id = widgetId;
+            this.setState(newState);
+        }
+
+        if (options) {
+            for (let i = 0; i < options.length; i++) {
+                switch (options[i].type) {
+                    case "TypeText":
+                        preferences.push(
+                            <div className="options-list">
+                                <TextField id={options[i].id}
+                                           floatingLabelText={options[i].title}
+                                           defaultValue={this.state.options[i].defaultData}
+                                           onChange={(event, newValue) => {
+                                               event.persist();
+                                               this.handlePreferenceTextBoxEvent(event, newValue, options)
```

Comment:
`handlePreferenceTextBoxEvent`
Seems like you are using "preference" & "option" words interchangeably. Lets stick to one, "option". Therefore above function should be renamed to `handleWidgetOptionTextFieldEvent`

Reply:
changed with 910980c413d9aea940c47a751f949593c1fc5ec0